lua/entities/sent_spawnpoint/init.lua

```diff
@@ -29,63 +34,60 @@ function unlinkPlayerFromSpawnPoint( player, spawnPoint )
-            end
-        end
+    if not IsValid( spawnPoint ) then return end
+
+    local linkedPlayers = spawnPoint.LinkedPlayers
+    local spawnPointOwner = spawnPoint:CPPIGetOwner()
+    
+    for player, _ in pairs( linkedPlayers ) do
+        if excludedPlayers[player] then next end
+        if spawnPointOwner == player then next end
+
+        unlinkPlayerFromSpawnPoint( player, spawnPoint)
+        player:PrintMessage(4, "You've been unlinked from a Spawn Point!")
     end
 end
 
 -- Chat commands
 function unlinkSpawnPointCommand( player, text, _, _ )
-    local text = string.lower( text ):gsub("%s+", "")
+    local text = string.lower( text ):gsub( "%s+", "" )
     local unlinkSpawnCommands = spawnPointCommands.unlinkSpawnPoint
 
-    if ( unlinkSpawnCommands[text] ) then
-        local linkedSpawnPoint = player.LinkedSpawnPoint
-        unlinkPlayerFromSpawnPoint( player, linkedSpawnPoint )
-        player:PrintMessage(4, "Spawn Point unlinked")
-    end
+    if not unlinkSpawnCommands[text] then return end
+
+    local linkedSpawnPoint = player.LinkedSpawnPoint
+    unlinkPlayerFromSpawnPoint( player, linkedSpawnPoint )
+    player:PrintMessage( 4, "Spawn Point unlinked" )
 end
 hook.Remove( "PlayerSay", "UnlinkSpawnPointCommand" )
 hook.Add( "PlayerSay", "UnlinkSpawnPointCommand", unlinkSpawnPointCommand )
 
 function unlinkThisSpawnPointCommand( player, text, _, _ )
-    local text = string.lower( text ):gsub("%s+", "")
+    local text = string.lower( text ):gsub( "%s+", "" )
     local unlinkThisSpawnCommands = spawnPointCommands.unlinkThisSpawnPoint
 
-    if ( unlinkThisSpawnCommands[text] ) then
-        local targetedEntity = player:GetEyeTraceNoCursor().Entity
-
-        if ( targetedEntity and targetedEntity:IsValid() ) then
-            local isSpawnPoint = targetedEntity:GetClass() == "sent_spawnpoint"
-
-            if ( isSpawnPoint ) then
-                local spawnPoint = targetedEntity
-                local spawnPointOwner = spawnPoint:CPPIGetOwner()
-                local playerOwnsSpawnPoint = spawnPointOwner == player
-                local playerIsAdmin = player:IsAdmin()
-
-                if ( playerOwnsSpawnPoint or playerIsAdmin ) then
-                    local excludedPlayers = createPlayerList( { spawnPointOwner } )
-                    unlinkAllPlayersFromSpawnPoint(spawnPoint, excludedPlayers)
-                    player:PrintMessage(4, "All players except the owner have been unlinked from this Spawn Point")
-                else
-                    player:PrintMessage(4, "That's not yours! You can't unlink others from this Spawn Point")
-                end
-            else
-                player:PrintMessage(4, "You must be looking at a Spawn Point to use this command")
-            end
-        end
+    if not unlinkThisSpawnCommands[text] then return end
+
+    local targetedEntity = player:GetEyeTraceNoCursor().Entity
+    if not targetedEntity or not targedtedEntity:IsValid() then return end
```

Comment:
Could also refactor to:

```lua
if not (targetedEntity and targetedEntity:IsValid()) then return end
```
I think this might be better just because it won't evaluate `targedEntity:IsValid()` unless `targetedEntity`